### ISSUE TITLE
Dates are serialized incorrectly for some namespaces

### ISSUE THREAD

**Issue by MJBonifacio** [helm/helm#7700]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.0.2", GitCommit:"19e47ee3283ae98139d98460de796c1be1e3975f", GitTreeState:"clean", GoVersion:"go1.13.5"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.10-eks-aae39f", GitCommit:"aae39f4697508697bf16c0de4a5687d464f4da81", GitTreeState:"clean", BuildDate:"2019-12-23T08:19:12Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
- AWS EKS

I'm seeing some strange behavior where helm ls will randomly output the date format for all releases in a namespace incorrectly:
```
➜  ~ helm ls --namespace reserved-8
NAME               NAMESPACE 	REVISION	UPDATED                               	STATUS  	CHART               	APP VERSION
elasticsearch	my-namespace	1       	2020-02-25 10:52:52.267935 -0700 -0700	deployed	elasticsearch-1.28.2	6.7.0
redis        	my-namespace	1       	2020-02-25 10:53:06.19247 -0700 -0700 	deployed	redis-8.0.8         	5.0.5
```

Where `2020-02-25 10:53:06.19247 -0700 -0700` should be something like `2020-02-25 10:53:06.19247 -0700 MST`.  

Any ideas what might be causing this? I see nothing in the helm code for serializing releases which could lead to this state.

**Comment by hickeyma** [comment]
This should be fixed by PR #7024 which is currently a WIP.

**Comment by technosophos** [comment]
This comes directly from Go's `time.Time.String()` function. My guess is that it cannot find the appropriate timezone files to figure out what the name of the TZ -0700 is.

**Comment by technosophos** [comment]
#7486 suggests that this is in some way related to Windows timestamps.

**Comment by bridgetkromhout** [comment]
After https://github.com/helm/helm/pull/7024 is released (currently on track for 3.4) we should be able to determine if this issue needs to remain open.
